extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/scan/ScanQueryKit.java

```diff
@@ -140,6 +141,20 @@ public QueryDefinition makeQueryDefinition(
       // Add partition boosting column.
       clusterByColumns.add(new KeyColumn(QueryKitUtils.PARTITION_BOOST_COLUMN, KeyOrder.ASCENDING));
       signatureBuilder.add(QueryKitUtils.PARTITION_BOOST_COLUMN, ColumnType.LONG);
+      final RowSignature signatureSoFar = signatureBuilder.build();
+      boolean addShuffle = true;
+      if (originalQuery.getContext().containsKey(MultiStageQueryContext.NEXT_WINDOW_SHUFFLE_COL)) {
+        final ClusterBy windowClusterBy = (ClusterBy) originalQuery.getContext().get(MultiStageQueryContext.NEXT_WINDOW_SHUFFLE_COL);
+        for (KeyColumn c : windowClusterBy.getColumns()) {
+          if (!signatureSoFar.contains(c.columnName())) {
+            addShuffle = false;
+            break;
+          }
+        }
+        if (addShuffle) {
+          clusterByColumns.addAll(windowClusterBy.getColumns());
```

Comment:
Here you are partition boosting and adding clustering columns. Won't stuff break ?

Reply:
Updated the code, need to discuss more on cases where there will be a window over a boosted scan column